bot/bot.py

```diff
@@ -72,6 +89,20 @@ async def close(self) -> None:
         if self.redis_session:
             await self.redis_session.close()
 
+    async def load_extension(
+        self,
+        name: str,
+        *,
+        package: Optional[str] = None,
+        ignore_cache: Optional[bool] = False
+    ) -> None:
+        """Load the extension only if it is not present in unloaded cache."""
+        if not ignore_cache and not await self.unloads_cache.get(name):
```

Comment:
This if-statement is still wrong, but now for the wrong reason. `await self.unloads_cache.get(name)` becomes None and then you invert this using `not`.

What ends up happening is the opposite of what we want.